external/js/cothority/src/skipchain/skipchain-rpc.ts

```diff
@@ -104,68 +106,54 @@ export default class SkipchainRPC {
     async getAllSkipChainIDs(): Promise<Buffer[]> {
         const req = new GetAllSkipChainIDs();
 
-        const ret = await this.conn[0].send<GetAllSkipChainIDsReply>(req, GetAllSkipChainIDsReply);
+        const ret = await this.pool.send<GetAllSkipChainIDsReply>(req, GetAllSkipChainIDsReply);
 
         return ret.skipChainIDs.map((id) => Buffer.from(id));
     }
 
     /**
      * Get the shortest path to the more recent block starting from latestID
      *
-     * @param latestID ID of the block
+     * @param latestID  ID of the block
+     * @param verify    Verify the integrity of the chain when true
      * @returns a promise that resolves with the list of blocks
      */
-    async getUpdateChain(latestID: Buffer): Promise<SkipBlock[]> {
+    async getUpdateChain(latestID: Buffer, verify = true): Promise<SkipBlock[]> {
         const req = new GetUpdateChain({ latestID });
-        const ret = await this.conn[0].send<GetUpdateChainReply>(req, GetUpdateChainReply);
+        const ret = await this.pool.send<GetUpdateChainReply>(req, GetUpdateChainReply);
+        const blocks = ret.update;
 
-        const err = this.verifyChain(ret.update, latestID);
-        if (err) {
-            throw new Error(`invalid chain received: ${err.message}`);
+        const last = blocks[blocks.length - 1];
+        if (last && last.forwardLinks.length > 0) {
+            // more blocks exist but typically the roster has changed
+            const rpc = new SkipchainRPC(last.roster);
```

Comment:
will this part create a lot of new connections? maybe some connections will be duplicated?

i don't know if this is a serious issue or if we need to do something about it because roster changes are infrequent.

Reply:
The connection is actually created only when you send something. Nothing is opened at this point.